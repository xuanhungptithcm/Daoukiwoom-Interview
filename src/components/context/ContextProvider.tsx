import { PropsWithChildren, ReactElement, useState } from "react";
import { IReactContext } from "./types";

interface IContextProviderProps<T> {
  Context: IReactContext<T>;
  initialState: T;
}

function ContextProvider<T>({
  Context,
  initialState,
  children,
}: PropsWithChildren<IContextProviderProps<T>>): ReactElement {
  const state = useState<T>(initialState);

  return <Context.Provider value={state}>{children}</Context.Provider>;
}

export default ContextProvider;
