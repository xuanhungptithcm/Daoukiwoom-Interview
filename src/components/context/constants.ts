import React from "react";
import { makeContext } from "./helpers";

import { IGeneralData } from "./types";

export const initialSetter: React.Dispatch<React.SetStateAction<unknown>> =
  // this is just a placeholder function, it will never be used
  () => {
    // EMPTY
  };

export const initialGeneralData: IGeneralData = {
  isLoggedIn: true,
  userInfo: {
    createdAt: "",
    name: "",
    avatar: "",
    username: "",
    password: "",
    id: "",
  },
};

export const GeneralDataContext = makeContext(initialGeneralData);
