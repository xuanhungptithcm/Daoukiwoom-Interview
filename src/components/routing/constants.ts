export enum AUTH_PATH {
  LOGIN_PAGE = "/login",
  AUTH_PAGE = "/auth",
}
