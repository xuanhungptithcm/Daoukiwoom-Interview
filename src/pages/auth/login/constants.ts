export const FormErrorMessage = {
  email: 'Email is invalid',
  password: `Password must At least One Upper Case Character \n At least one Lower Case character \n At least one digit \n At least one symbol/special character @$!%*#?&^_- \n Minimum 8 characters/digits`
}