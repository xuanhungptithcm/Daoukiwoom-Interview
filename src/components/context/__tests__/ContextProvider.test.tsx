/**
 * @jest-environment jsdom
 */

import { shallow } from "enzyme";
import { GeneralDataContext } from "../constants";
import ContextProvider from "../ContextProvider";

describe("ContextProvider", () => {
  test("should properly render", () => {
    expect(
      shallow(
        <ContextProvider Context={GeneralDataContext} initialState={{} as any}>
          children
        </ContextProvider>
      )
    ).toMatchSnapshot();
  });
});
