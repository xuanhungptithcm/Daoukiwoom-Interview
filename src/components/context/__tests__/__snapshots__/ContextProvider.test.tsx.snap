// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextProvider should properly render 1`] = `
<ContextProvider
  value={
    Array [
      Object {},
      [Function],
    ]
  }
>
  children
</ContextProvider>
`;
